src/overlays/actors/ovl_Obj_Roomtimer/z_obj_roomtimer.c

```diff
@@ -31,16 +30,62 @@ const ActorInit Obj_Roomtimer_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjRoomtimer_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjRoomtimer* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Roomtimer/ObjRoomtimer_Init.s")
+    this->flag = ROOMTIMER_GET_SWITCHFLAG(thisx);
+    this->actor.params &= 0x1FF;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Roomtimer/ObjRoomtimer_Destroy.s")
+    if (this->actor.params != 0x1FF) {
+        this->actor.params = CLAMP_MAX(this->actor.params, 0x1F4);
```

Comment:
I think this is dec, since it's used as the time limit?
```suggestion
        this->actor.params = CLAMP_MAX(this->actor.params, 600);
```